File: backend/src/main/java/com/example/backend/entity/Quiz.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Entity
@Data
@Table(
        name = "quizzes",
        indexes = {
                @Index(name = "idx_quizzes_creator", columnList = "creator_id"),
                @Index(name = "idx_quizzes_category", columnList = "category_id"),
        }
)
public class Quiz {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "creator_id", nullable = false)
    private User creator;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(columnDefinition = "TEXT")
    private String summary;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private SourceType source_type = SourceType.TEXT;

    @ManyToOne
    @JoinColumn(name = "file_id", nullable = true)
    private UploadedFile  file;

    @OneToMany(mappedBy = "quiz", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Question> questions;
}

File: my-app/app/Models/Quiz.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Quiz extends Model
{
    use HasFactory;

    public $timestamps = false;

    protected $fillable = [
        'creator_id',
        'category_id',
        'title',
        'description',
        'summary',
        'source_type',
        'file_id',
        'avatar',
    ];

    public function creator()
    {
        return $this->belongsTo(User::class, 'creator_id');
    }

    public function category()
    {
        return $this->belongsTo(Category::class, 'category_id');
    }

    public function file()
    {
        return $this->belongsTo(UploadedFile::class, 'file_id');
    }

    public function questions()
    {
        return $this->hasMany(Question::class)->orderBy('order_index');
    }

    public function finalResults()
    {
        return $this->hasMany(FinalResult::class);
    }
}